Update NZBGet to 15.0

DaftHonk opened the pull request:
Built and tested working on cedarview-5.1 (DS1813+)

Diaoul commented:
Changelog is missing
Default configuration file is broken
No explanations on other changes

Dr-Bean commented:
Popular package, it's the third PR already ;) I was also working on it.
Apart from @Diaoul's comments, what I'm puzzled about is the addition of `cross/p7zip`. @DaftHonk, what's that for?

DaftHonk commented:
For the dependencies I was going off what was in the nzbget-testing Makefile - I made the assumption that p7zip was added in v15.0. I will have to try compiling without and confirm if it is required. I will add in the other missing items and fix the config file.

Dr-Bean commented:
The default configuration uses Synology's 7z (`/usr/syno/bin/7z`), so unless one changes the path after package installation, the compiled binary simply won't get used.
You might as well change the path in the config to make sure it does get used, and while you're at it, the same goes for `unrar`.

Dr-Bean commented:
Correction: unless a `7z` binary is found in `/usr/local/nzbget/bin`, Synology's binary is used...we set the correct PATH in `dsmcontrol.sh`.

DaftHonk commented on spk/nzbget/Makefile:
```diff
@@ -49,9 +49,7 @@ nzbget_extra_install: $(STAGING_DIR)/share/nzbget/scripts/nzbToMedia $(STAGING_D
 
 $(STAGING_DIR)/share/nzbget/scripts/nzbToMedia:
 	cd $(STAGING_DIR)/share/nzbget/scripts && git clone https://github.com/clinton-hall/nzbToMedia.git
-	rm -fr $(STAGING_DIR)/share/nzbget/scripts/nzbToMedia/.git*
```
removed to ensure auto-update on script execution works

Dr-Bean replied:
As long as it actually works. Did you verify that?

DaftHonk replied:
I confirmed that this works:

INFO	Mon Jun 01 2015 23:06:17	nzbToNzbDrone: [23:06:17] [INFO]::MAIN: No update needed
INFO	Mon Jun 01 2015 23:06:17	nzbToNzbDrone: [23:06:17] [INFO]::MAIN: Checking if git needs an update
INFO	Mon Jun 01 2015 23:06:17	nzbToNzbDrone: [23:06:17] [INFO]::MAIN: Checking database structure...
INFO	Mon Jun 01 2015 23:06:15	nzbToNzbDrone: [23:06:15] [INFO]::MAIN: Restarting nzbToMedia with ['/usr/local/python/bin/python2', '/usr/local/nzbget/share/nzbget/scripts/nzbToMedia/nzbToNzbDrone.py']
INFO	Mon Jun 01 2015 23:06:12	nzbToNzbDrone: [23:06:12] [INFO]::MAIN: Auto-Updating nzbToMedia, Please wait ...
INFO	Mon Jun 01 2015 23:06:12	nzbToNzbDrone: [23:06:12] [INFO]::MAIN: There is a newer version available (you're 1 commit behind)
INFO	Mon Jun 01 2015 23:06:11	nzbToNzbDrone: [23:06:11] [INFO]::MAIN: Checking if git needs an update

DaftHonk commented on spk/nzbget/Makefile:
```diff
@@ -49,9 +49,7 @@ nzbget_extra_install: $(STAGING_DIR)/share/nzbget/scripts/nzbToMedia $(STAGING_D
 
 $(STAGING_DIR)/share/nzbget/scripts/GetScripts:
 	cd $(STAGING_DIR)/share/nzbget/scripts && git clone https://github.com/clinton-hall/GetScripts.git
-	rm -fr $(STAGING_DIR)/share/nzbget/scripts/GetScripts/.git*
```
removed to ensure auto-update on script execution works

DaftHonk commented:
I updated the config file to use the latest template from version 15.0, but added back in the necessary arguments for the Synology package. I deleted cross/nzbget/patches/rss-content-length.patch because it failed to compile on my system.

Dr-Bean commented on spk/nzbget/Makefile:
```diff
@@ -1,6 +1,6 @@
-SPK_VERS = 14.2
-SPK_REV = 15
+SPK_VERS = 15.0
+SPK_REV = 1
```
16. The SPK revision only goes up, not down.

Dr-Bean commented on spk/nzbget/src/nzbget.conf:
```diff
@@ -85,8 +85,14 @@ ScriptDir=/usr/local/nzbget/share/nzbget/scripts
 
```
File has changed permissions. Any particular reason? Otherwise best to revert that.

Diaoul replied:
I don't remember why it's 755 but I believe it should be 644.
That doesn't change anything anyway because the file is installed with 644: https://github.com/SynoCommunity/spksrc/blob/master/spk/nzbget/Makefile#L39

Dr-Bean replied:
Sure :)

DaftHonk replied:
This file was copied directly from the tarball for 15.0. It must have preserved its permissions from there - I will fix that.

Dr-Bean commented:
I was considering changing the default logging settings to rotate (`WriteLog=rotate`), and keep the other settings as default (`ErrorTarget=both`, `WarningTarget=both`, `InfoTarget=both`, `DetailTarget=log` and although there's no direct effect: `DebugTarget=log`).
Setting `rotate` keeps the logfile limited in size (current default is append), and keeping the other settings to default makes things easier to maintain.
What do you think?

Diaoul commented:
OK for me

DaftHonk commented:
Your log suggestions are the same that I use on my system and it indeed keeps log size down. I'm not sure how it will work with the new per-nzb logging feature.

Dr-Bean commented:
The per-nzb logging feature isn't a logfile per nzb, afaik, but it's a filter of the logs.

Dr-Bean commented:
Package nzbget 15.0-16 has been published
